src/schema/validator.js

```diff
@@ -241,42 +323,48 @@ export class SchemaValidator {
     return this._localeValidator;
   }
 
-  _compileAddonValidator(validator, validatorOptions) {
-    const { minManifestVersion, maxManifestVersion } = validatorOptions;
-
-    let schemaData = this.schemaObject;
-
-    if (minManifestVersion != null || maxManifestVersion != null) {
-      const minimum =
-        minManifestVersion == null
-          ? getDefaultConfigValue('min-manifest-version')
-          : minManifestVersion;
-      const maximum =
-        maxManifestVersion == null
-          ? getDefaultConfigValue('max-manifest-version')
-          : maxManifestVersion;
-
-      // Make sure the version range is valid, if it is not:
-      // raise an explicit error.
-      if (maxManifestVersion < minManifestVersion) {
-        throw new Error(
-          `Invalid manifest version range requested: ${JSON.stringify({
-            maxManifestVersion,
-            minManifestVersion,
-          })}`
-        );
-      }
-
-      schemaData = deepPatch(this.schemaObject, {
-        types: {
-          ManifestBase: {
-            properties: {
-              manifest_version: { minimum, maximum },
+  get allowedManifestVersionsRange() {
```

Comment:
When should we prefer a getter over a function? I don't think this necessarily needs to be a (computed) property.

Reply:
yeah, honestly I think that I agree with you, this may be actually computed once in the constructor (especially given that it is also validating the resulting range and throwing an error if it turns out to be invalid).

I'll apply that change and update this PR accordingly.